Cookbook/Technical-Documents/FeatureSwitches.md

```diff
@@ -0,0 +1,117 @@
+# Feature Switches
+
+This document describes the process that should be followed when defining Features Switches.
+
+Depending on the use case we are using different ways to define Feature Switches in our code base. Feature Switches is not always some `Bool` value that can be either `true` or `false`, though most of the times it will be just a `Bool`, it can be a more complicated type that defines a conditional behaviour.
+
+1. **Local feature switches**
+
+	These feature switches are defined in the `LocalFeatureSwitchesProtocol` and are stored in `UserDefaults`. **These feature switches are supposd to be used only during the development of any new feature so that the code for the feature can be integrated in the `develop` branch continuously.** As they are stored in `UserDefaults` they can be overriden during testing with Settings app. To add a new local feature switch do following:
+
+	1. add a new case in the `SettingsKeys` enum for this feature flag in `LocalFeatureSwitches.swift`:
+
+	```swift
+	case isNewFeatureEnabled
+	```	
+	
+	2. add a new property to the `LocalFeatureSwitchesProtocol` in `LocalFeatureSwitches.swift`:
+
+	```swift
+	static var isNewFeatureEnabled: Bool { get }
+	```
+	
+	3. add a default implementation of a feature switch property in the extension of `LocalFeatureSwitchesProtocol` in `LocalFeatureSwitches.swift`:
+
+	```swift
+    public static var isNewFeatureEnabled: Bool {
+        #if DEBUG || ENABLE_FEATURE_FLAGS
+        return valueFromDefaults(for: .isNewFeatureEnabled)
+        #else
+        return false
+        #endif
+    }
+	```
+	
+	Note: `ENABLE_FEATURE_FLAGS` compilation codition exists so that we are able to disable all the **local feature flags** in the builds that we provide for user testing. This flag should be used **only with local feature switches**.
```

Comment:
```suggestion
	Note: `ENABLE_FEATURE_FLAGS` compilation condition exists so that we are able to disable all the **local feature flags** in the builds that we provide for user testing. This flag should be used **only with local feature switches**.
```